### PR TITLE
Enterprise GitHub

### PR DESCRIPTION
**Pull request by bonzofenix**
Merged https://github.com/billimek/tracksuit Enterprise GitHub feature only. Fixes conflicts with current branch.

### CODE REVIEW

**Comment by billimek** [comment]
Would love to see this merged, @vito!

File: syncer.go [hunk 1]
```diff
@@ -48,7 +48,8 @@ type Syncer struct {
 	ProjectClient tracker.ProjectClient
 
 	OrganizationName string
-	Repositories     []string
+
```
Comment (vito) [inline syncer.go]: unrelated change; could you clean this up?

File: scripts/run [hunk 2]
```diff
@@ -10,10 +10,11 @@ go install github.com/vito/tracksuit
 export TRACKSUIT_GITHUB_TOKEN=${TRACKSUIT_GITHUB_TOKEN:-$GITHUB_TOKEN}
 export TRACKSUIT_GITHUB_ORGANIZATION_NAME=${TRACKSUIT_GITHUB_ORGANIZATION_NAME:-$ORGANIZATION}
 export TRACKSUIT_GITHUB_REPOSITORIES=${TRACKSUIT_GITHUB_REPOSITORIES:-$REPOSITORIES}
+export TRACKSUIT_GITHUB_API_URL=${TRACKSUIT_GITHUB_API_URL:-$GITHUB_API_URL}
 export TRACKSUIT_TRACKER_TOKEN=${TRACKSUIT_TRACKER_TOKEN:-$TRACKER_TOKEN}
 export TRACKSUIT_TRACKER_PROJECT_ID=${TRACKSUIT_TRACKER_PROJECT_ID:-$PROJECT_ID}
 export TRACKSUIT_GC_LABELS=${TRACKSUIT_GC_LABELS:-$GC_LABELS}
 
-unset GITHUB_TOKEN ORGANIZATION REPOSITORIES TRACKER_TOKEN PROJECT_ID GC_LABELS
+unset GITHUB_API_URL GITHUB_TOKEN ORGANIZATION REPOSITORIES TRACKER_TOKEN PROJECT_ID GC_LABELS
```
Comment (vito) [inline scripts/run]: don't need this; these vars are only there for backwards compatibility

File: main.go [hunk 3]
```diff
@@ -30,13 +33,28 @@ type TracksuitCommand struct {
 	GCLabels bool `long:"gc-labels" description:"Garbage collect labels in Tracker that no longer reference an issue"`
 }
 
+var repositories = NewStringSet()
```
Comment (vito) [inline main.go]: this and the function below it should be removed

File: main.go [hunk 4]
```diff
@@ -30,13 +33,28 @@ type TracksuitCommand struct {
...
+
 func (cmd *TracksuitCommand) Execute(argv []string) error {
 	ghToken := &oauth2.Token{AccessToken: cmd.GitHub.Token}
 
 	ghAuth := oauth2.NewClient(oauth2.NoContext, oauth2.StaticTokenSource(ghToken))
 
 	githubClient := github.NewClient(ghAuth)
 
+	if cmd.GitHub.ApiUrl != "" {
+		var Url *url.URL
```
Comment (vito) [inline main.go]: vars should be lowercased

File: main.go [hunk 5]
```diff
@@ -30,13 +33,28 @@ type TracksuitCommand struct {
...
+		Url, _ = url.Parse(cmd.GitHub.ApiUrl)
```
Comment (vito) [inline main.go]: check err here; it shouldn't be ignored. just bubble it up (`return err`)

File: main.go [hunk 6]
```diff
@@ -3,25 +3,28 @@ package main
...
-		Repositories []string `long:"repository" desciption:"Repository to sync. Can be repeated to sync many repositories. If omitted, all repositories are synced."`
+		Token            string   `long:"token"	required:"true" description:"GitHub access token"`
+		OrganizationName string   `long:"organization-name" required:"true" description:"GitHub organization name"`
+		Repositories     []string `long:"repository" desciption:"Repository to sync. Can be repeated to sync many repositories. If omitted, all repositories are synced."`
+		ApiUrl           string   `long:"api-url" description:"Github api url. If omitted it defaults to api.github.com"`
 	} `group:"GitHub Configuration" namespace:"github"`
 
 	Tracker struct {
-		Token     string `long:"token"      required:"true" description:"Tracker Access token"`
+		Token     string `long:"token"			required:"true" description:"Tracker Access token"`
```
Comment (vito) [inline main.go]: what happened with the spacing here? can you undo this?

File: main.go [hunk 7]
```diff
@@ -3,25 +3,28 @@ package main
...
 type TracksuitCommand struct {
 	GitHub struct {
-		Token            string `long:"token"             required:"true" description:"GitHub access token"`
-		OrganizationName string `long:"organization-name" required:"true" description:"GitHub organization name"`
-
-		Repositories []string `long:"repository" desciption:"Repository to sync. Can be repeated to sync many repositories. If omitted, all repositories are synced."`
+		Token            string   `long:"token"	required:"true" description:"GitHub access token"`
+		OrganizationName string   `long:"organization-name" required:"true" description:"GitHub organization name"`
+		Repositories     []string `long:"repository" desciption:"Repository to sync. Can be repeated to sync many repositories. If omitted, all repositories are synced."`
+		ApiUrl           string   `long:"api-url" description:"Github api url. If omitted it defaults to api.github.com"`
```
Comment (vito) [inline main.go]: initialisms are usually capitalized (e.g. `APIURL`)

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,3 @@
+FROM golang:1.6
```
Comment (vito) [inline Dockerfile]: let's not add this; seems unrelated

**Comment by billimek** [comment]
@vito these changes should be present in a new pull request, #13
